src/cc65/scanner.c

```diff
@@ -1267,7 +1267,8 @@ static int CloseBrace (Collection* C, token_t Tok)
 */
 {
     if (CollCount (C) > 0) {
-        token_t LastTok = (token_t)CollLast (C);
+        token_t* LastTok_ptr = (token_t*)CollLast (C);
+        token_t LastTok = *LastTok_ptr;
```

Comment:
Have you checked that this code actually works? You don't just calmed down the warning, you changed the logic:

1. Old code: reinterprets the CollLast(c) as token_t LastTok
2. New code: dereferences (!) the CollLast(c), which is now considered to be a pointer to token_t

This is a very essential change. If the change is right, the old code was actually wrong. Otherwise, you risk accessing an invalid pointer.